### PR TITLE
[IMP] web_tour: add a sequence option to tours

### PR DESCRIPTION
**Pull request by Arcasias**
Before this commit, tours were activated in a non deterministic way and there was
no way to control the order.

Now there is a `sequence` key in the options dict (i.e. the second parameters of
`tour.register`) which will determine the tour sequence order, the lowest being
first and going upward. Unsequenced tours will behave as before with no particular
order.

Also added a sequence on a few tours.

Task 2191192

Enterprise: https://github.com/odoo/enterprise/pull/9472
Closes https://github.com/odoo/odoo/issues/33405

### CODE REVIEW

File: addons/web_tour/static/src/js/tour_manager.js [hunk 1]
```diff
@@ -51,6 +54,9 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
      *        the url to load when manually running the tour
      * @param {boolean} [options.rainbowMan=true]
      *        whether or not the rainbowman must be shown at the end of the tour
+     * @param {boolean} [options.sequence]
+     *        priority sequence of the tour (lowest is first, no sequence = comes
+     *        after other sequenced tours in a non deterministic order).
```
Comment (aab-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: not sure about this. I would force a default sequence (like 50), so that someone that specify an higher sequence for his tour, ensuring it will be activated *after* the others, even those with no sequence

Comment (aab-odoo) [reply addons/web_tour/static/src/js/tour_manager.js]: That's what we did for the Systray items, for instance

File: addons/web_tour/static/src/js/tour_manager.js [hunk 2]
```diff
@@ -295,7 +305,9 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
         if (this.running_tour !== tour_name) {
             tip.widget.on('tip_consumed', this, this._consume_tip.bind(this, tip, tour_name));
         }
-        tip.widget.attach_to($anchor).then(this._to_next_running_step.bind(this, tip, tour_name));
+        tip.widget.attach_to($anchor)
+            .then(this._to_next_running_step.bind(this, tip, tour_name))
+            .then(this.setTooltipReady);
```
Comment (aab-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: why did you need this?

Comment (Arcasias) [reply addons/web_tour/static/src/js/tour_manager.js]: This was essentially to avoid calling needless `testUtils.nextTicks` in the tests, but it could be used in the future as an assertion that the tour manager and its widgets are fully ready.

File: addons/web_tour/static/src/js/tour_manager.js [hunk 3]
```diff
@@ -51,6 +54,9 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
      *        the url to load when manually running the tour
      * @param {boolean} [options.rainbowMan=true]
      *        whether or not the rainbowman must be shown at the end of the tour
+     * @param {boolean} [options.sequence=1000]
+     *        priority sequence of the tour (lowest is first, no sequence = comes
+     *        after other sequenced tours in a non deterministic order).
```
Comment (aab-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: message doesn't correspond to implementation anymore

File: addons/web_tour/static/src/js/tour_manager.js (hunk 2 above)
Comment (aab-odoo) [reply addons/web_tour/static/src/js/tour_manager.js]: Everything should be sync. If it's not, I guess it's because of a settimeout. You can parameterize the timeout value and override it in the test like we often do.

Comment (aab-odoo) [reply addons/web_tour/static/src/js/tour_manager.js]: I don't really see a usecase where we would need this, and it makes the code a bit more complicated, so if we can avoid this, I think it's better.

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r-

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo r+
